### ISSUE TITLE
Cata Classic - script ran too long

### ISSUE THREAD

**Issue by theoriginalmre** [Auctionator/Auctionator#1547]
Tried to sell 4 items and got this error.
Then tried to do a Full Scan, and it got to 99% and gave me this error.
(Using 11.0.3-cata)

1x Auctionator/Source/Database/Mixin.lua:104: script ran too long
[string "@Auctionator/Source/Database/Mixin.lua"]:104: in function `SetPrice'
[string "@Auctionator/Source_Classic/Tabs/Selling/Mixins/SaleItem.lua"]:754: in function `callback'
[string "@Auctionator/Source/Utilities/DBKeyFromLink.lua"]:59: in function `DBKeyFromLink'
[string "@Auctionator/Source_Classic/Tabs/Selling/Mixins/SaleItem.lua"]:752: in function `SuccessfulPost'
[string "@Auctionator/Source_Classic/Tabs/Selling/Mixins/SaleItem.lua"]:361: in function `ReceiveEvent'
[string "@Auctionator/Source/Utilities/EventBusMixin.lua"]:77: in function `Fire'
[string "@Auctionator/Source_Classic/Tabs/Selling/Mixins/PostWatch.lua"]:36: in function <...tor/Source_Classic/Tabs/Selling/Mixins/PostWatch.lua:28>

Locals:
self = <table> {
 GetFirstPrice = <function> defined @Auctionator/Source/Database/Mixin.lua:131
 GetMeanPrice = <function> defined @Auctionator/Source/Database/Mixin.lua:229
 Init = <function> defined @Auctionator/Source/Database/Mixin.lua:8
 ProcessScan = <function> defined @Auctionator/Source/Database/Mixin.lua:143
 GetPriceHistory = <function> defined @Auctionator/Source/Database/Mixin.lua:179
 processor = Frame {
 }
 GetPrice = <function> defined @Auctionator/Source/Database/Mixin.lua:123
 cutoffDay = 1666
 SetPrice = <function> defined @Auctionator/Source/Database/Mixin.lua:53
 db = <table> {
 }
 GetPriceAge = <function> defined @Auctionator/Source/Database/Mixin.lua:207
 _Queue = <function> defined @Auctionator/Source/Database/Mixin.lua:18
 GetItemCount = <function> defined @Auctionator/Source/Database/Mixin.lua:170
}
dbKey = "52327"
buyoutPrice = 19997
available = nil
priceData = <table> {
 a = <table> {
 }
 l = <table> {
 }
 h = <table> {
 }
 m = 19997
}
lowestLow = nil
highestLow = 19998
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 1677 = 20836
 1665 = 6664
 1681 = 14117
 1671 = 10571
 1685 = 19875
 1666 = 28696
 1679 = 26787
 1680 = 22195
 1675 = 11397
 1687 = 19997
 1686 = 32990
 1676 = 25984
 1674 = 11000
 1673 = 21950
 1678 = 11317
}
(for control) = "1671"
day = "1671"
_ = 10571
(*temporary) = 1671
(*temporary) = "1671"
(*temporary) = "script ran too long"
get = <function> defined @Auctionator/Source/Database/Mixin.lua:45
daysSinceZero = "1687"

**Comment by danjohnso** [comment]
Also occurs in SOD.  Same behavior with trying to search an item, putting an item into sell, or doing a full scan:

```Message: ...terface/AddOns/Auctionator/Source/Database/Mixin.lua:99: script ran too long
Time: Wed Aug 14 08:26:26 2024
Count: 1
Stack: ...terface/AddOns/Auctionator/Source/Database/Mixin.lua:99: script ran too long
[string "=(tail call)"]: ?
[string "@Interface/AddOns/Auctionator/Source/Database/Mixin.lua"]:99: in function `SetPrice'
[string "@Interface/AddOns/Auctionator/Source/Database/Mixin.lua"]:163: in function `ProcessScan'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/GroupResultsForDB.lua"]:11: in function <...tionator/Source_Classic/Search/GroupResultsForDB.lua:9>
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/GroupResultsForDB.lua"]:33: in function `callback'
[string "@Interface/AddOns/Auctionator/Source/Utilities/DBKeyFromLink.lua"]:59: in function `DBKeyFromLink'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/GroupResultsForDB.lua"]:20: in function `GroupResultsForDB'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:132: in function <..._Classic/Search/Mixins/DirectSearchProviderMixin.lua:123>
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:156: in function <..._Classic/Search/Mixins/DirectSearchProviderMixin.lua:149>
[string "=[C]"]: in function `xpcall'
[string "@Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua"]:298: in function `FireCallbacks'
[string "@Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua"]:260: in function <Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua:256>
[string "=[C]"]: ?
[string "=[C]"]: in function `RequestLoadItemDataByID'
[string "@Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua"]:274: in function `AddCallback'
[string "@Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua"]:238: in function `ContinueOnItemLoad'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:149: in function `AddFinalResults'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:182: in function `ProcessSearchResults'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:193: in function `ReceiveEvent'
[string "@Interface/AddOns/Auctionator/Source/Utilities/EventBusMixin.lua"]:77: in function `Fire'
[string "@Interface/AddOns/Auctionator/Source_Classic/AH/Mixins/Scan.lua"]:100: in function `ProcessSearchResults'
[string "@Interface/AddOns/Auctionator/Source_Classic/AH/Mixins/Scan.lua"]:39: in function <...AddOns/Auctionator/Source_Classic/AH/Mixins/Scan.lua:36>
Locals: (*temporary) = <function> defined =[C]:-1
```

**Comment by plusmouse** [comment]
Did you get a 10s+ freeze?

**Comment by theoriginalmre** [comment]
Yes whole game froze for ages

**Comment by plusmouse** [comment]
How many items are usually scanned when you do a full scan?

**Comment by theoriginalmre** [comment]
> How many items are usually scanned when you do a full scan?

Scan gets to 99% and never completed

**Comment by plusmouse** [comment]
Can you try this (Cata)
[Auctionator-11.0.3-1-ga22b903-cata.zip](https://github.com/user-attachments/files/16614278/Auctionator-11.0.3-1-ga22b903-cata.zip)
Or this (SoD) and report if it works?
[Auctionator-11.0.3-1-ga22b903-classic.zip](https://github.com/user-attachments/files/16614280/Auctionator-11.0.3-1-ga22b903-classic.zip)

**Comment by danjohnso** [comment]
Works in SOD

**Comment by danjohnso** [comment]
Well I spoke too soon.  The scan completes but then switching tabs caused another 10s stall with a shorter stack trace:
```
Message: ...terface/AddOns/Auctionator/Source/Database/Mixin.lua:99: script ran too long
Time: Wed Aug 14 08:26:26 2024
Count: 1
Stack: ...terface/AddOns/Auctionator/Source/Database/Mixin.lua:99: script ran too long
[string "=(tail call)"]: ?
[string "@Interface/AddOns/Auctionator/Source/Database/Mixin.lua"]:99: in function `SetPrice'
[string "@Interface/AddOns/Auctionator/Source/Database/Mixin.lua"]:163: in function `ProcessScan'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/GroupResultsForDB.lua"]:11: in function <...tionator/Source_Classic/Search/GroupResultsForDB.lua:9>
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/GroupResultsForDB.lua"]:33: in function `callback'
[string "@Interface/AddOns/Auctionator/Source/Utilities/DBKeyFromLink.lua"]:59: in function `DBKeyFromLink'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/GroupResultsForDB.lua"]:20: in function `GroupResultsForDB'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:132: in function <..._Classic/Search/Mixins/DirectSearchProviderMixin.lua:123>
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:156: in function <..._Classic/Search/Mixins/DirectSearchProviderMixin.lua:149>
[string "=[C]"]: in function `xpcall'
[string "@Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua"]:298: in function `FireCallbacks'
[string "@Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua"]:260: in function <Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua:256>
[string "=[C]"]: ?
[string "=[C]"]: in function `RequestLoadItemDataByID'
[string "@Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua"]:274: in function `AddCallback'
[string "@Interface/AddOns/Blizzard_ObjectAPI/Classic/Item.lua"]:238: in function `ContinueOnItemLoad'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:149: in function `AddFinalResults'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:182: in function `ProcessSearchResults'
[string "@Interface/AddOns/Auctionator/Source_Classic/Search/Mixins/DirectSearchProviderMixin.lua"]:193: in function `ReceiveEvent'
[string "@Interface/AddOns/Auctionator/Source/Utilities/EventBusMixin.lua"]:77: in function `Fire'
[string "@Interface/AddOns/Auctionator/Source_Classic/AH/Mixins/Scan.lua"]:100: in function `ProcessSearchResults'
[string "@Interface/AddOns/Auctionator/Source_Classic/AH/Mixins/Scan.lua"]:39: in function <...AddOns/Auctionator/Source_Classic/AH/Mixins/Scan.lua:36>

Locals: (*temporary) = <function> defined =[C]:-1
```

**Comment by plusmouse** [comment]
Can you try this build in SoD?
[Auctionator-11.0.3-2-gd3f1742-classic.zip](https://github.com/user-attachments/files/16614506/Auctionator-11.0.3-2-gd3f1742-classic.zip)

**Comment by danjohnso** [comment]
Seems like its good, no errors or freezing

**Comment by plusmouse** [comment]
Awesome. I've pushed the fix into 11.0.4, this will be available shortly. Closing this issue as completed :)
